### PR TITLE
Create resource timing entries for fetch network errors

### PR DESCRIPTION
**Pull request by noamr**
An initial tests for https://github.com/whatwg/fetch/issues/1215

Added test cases for several network error scenarios. The test cases verify that a resource timing entry is created for those resources, without the "internal" timing info (redirects, networking), as if the resource was cross-origin no-cors and lacked TAO.

The error test cases include:
- CORS fail (which is also covered in existing TAO tests)
- about URLs
- invalid blobs
- Posting to blobs
- Only-if-cached header for a fresh resource
- Abort HTTP/blob request
- Invalid scheme
- Too many redirects
- Invalid certificate
- Host not found
- CSP blocked

These tests come in conjunction with https://github.com/whatwg/fetch/pull/1311

### CODE REVIEW

File: resource-timing/resources/entry-invariants.js
Comment (yoavweiss) [inline resource-timing/resources/entry-invariants.js:499]: Should that be "greater_than_equal"?

**Review by yoavweiss — COMMENTED** [review]
Thanks!! A few comments

File: resource-timing/csp-block-fetch.js.headers
Comment (noamr) [reply resource-timing/csp-block-fetch.js.headers:1]: Right, debris from copy/paste

**Review by yoavweiss — APPROVED** [review]
LGTM

**Comment by noamr** [comment]
> LGTM

Thanks!
Should probably wait until the spec change is approved and merge at the same time.

**Comment by noamr** [comment]
Modified test to include only 5 scenarios for now:
- CORS failure
- DNS failure
- Timeout (IP address not found)
- Only-if-cached-resource that wasn't cached
- Too many redirects

Currently Chrome & Firefox are inconsistent around which of these create a resource timing entry.
Safari does not create RT entries for network errors.

**Comment by noamr** [comment]
This actually catches a Firefox bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1523275

File: resource-timing/entries-for-network-errors.sub.https.html
Comment (yoavweiss) [inline resource-timing/entries-for-network-errors.sub.https.html:31]: Mixed content could be another interesting case

File: resource-timing/entries-for-network-errors.sub.https.html
Comment (yoavweiss) [inline resource-timing/entries-for-network-errors.sub.https.html:24]: Nit: ";"

**Review by yoavweiss — APPROVED** [review]
LGTM % nit and suggestion for another test

**Comment by noamr** [comment]
The Firefox failures are legit, e.g. https://bugzilla.mozilla.org/show_bug.cgi?id=1523275. How does this get merged when it fails firefox nightly?

**Comment by noamr** [comment]
> LGTM % nit and suggestion for another test

Added a test for mixed content.

**Pull request merged**
